tests/e2e/test_e2e_large_trees.py

```diff
@@ -0,0 +1,107 @@
+import sys
+import lightgbm
+import pytest
+import numpy as np
+import xgboost
+from sklearn import linear_model, svm
+from sklearn import tree
+from sklearn import ensemble
+
+from tests import utils
+from tests.e2e import executors
+
+
+RECURSION_LIMIT = 5000
+
+
+# pytest marks
+PYTHON = pytest.mark.python
+JAVA = pytest.mark.java
+C = pytest.mark.c
+GO = pytest.mark.go
+JAVASCRIPT = pytest.mark.javascript
+REGRESSION = pytest.mark.regr
+CLASSIFICATION = pytest.mark.clf
+
+
+# Set of helper functions to make parametrization less verbose.
+def regression(model):
+    return (
+        model,
+        utils.train_model_regression_random_data,
+        REGRESSION,
+    )
+
+
+def classification(model):
+    return (
+        model,
+        utils.train_model_classification_random_data,
+        CLASSIFICATION,
+    )
+
+
+def classification_binary(model):
+    return (
+        model,
+        utils.train_model_classification_binary_random_data,
+        CLASSIFICATION,
+    )
+
+
+# Absolute tolerance. Used in np.isclose to compare 2 values.
+# We compare 6 decimal digits.
+ATOL = 1.e-6
+
+RANDOM_SEED = 1234
+XGBOOST_PARAMS = dict(base_score=0.6, n_estimators=100, max_depth=12,
+                      random_state=RANDOM_SEED)
+LIGHT_GBM_PARAMS = dict(n_estimators=100, num_leaves=100, max_depth=64,
+                    random_state=RANDOM_SEED)
+
+
+@utils.cartesian_e2e_params(
+    # These are the languages which support all models specified in the
+    # next list.
+    [
+        (executors.PythonExecutor, PYTHON),
+        (executors.JavaExecutor, JAVA),
+        (executors.CExecutor, C),
+        (executors.GoExecutor, GO),
+        (executors.JavascriptExecutor, JAVASCRIPT),
+    ],
```

Comment:
I think this also can be factored out into `e2e/__init__.py` as a `ALL_EXECUTORS` or something like that.